src/launchpad/kafka.py

```diff
@@ -205,39 +204,48 @@ def is_healthy(self) -> bool:
 class LaunchpadStrategyFactory(ProcessingStrategyFactory[KafkaPayload]):
     """Factory for creating the processing strategy chain."""
 
-    @staticmethod
-    def _initialize_worker_logging() -> None:
-        """Initialize logging in worker process.
-
-        With multiprocessing spawn context, subprocesses don't inherit
-        parent's stdout/stderr. We need to explicitly configure logging
-        to write to stdout for Docker/GCP log collection.
-        """
-        logging.getLogger().handlers.clear()
-
-        # Ensure stdout/stderr are unbuffered for immediate log visibility
-        if hasattr(sys.stdout, "reconfigure"):
-            sys.stdout.reconfigure(line_buffering=True)  # type: ignore[attr-defined]
-        if hasattr(sys.stderr, "reconfigure"):
-            sys.stderr.reconfigure(line_buffering=True)  # type: ignore[attr-defined]
-
-        server_config = get_server_config()
-        setup_logging(verbose=server_config.debug, quiet=not server_config.debug)
-
     def __init__(
         self,
         concurrency: int,
         max_pending_futures: int,
         healthcheck_file: str | None = None,
     ) -> None:
+        self._log_queue: multiprocessing.Queue[Any] = multiprocessing.Manager().Queue(-1)
+        self._queue_listener = self._setup_queue_listener()
+        self._queue_listener.start()
+
+        initializer_with_queue = partial(self._initialize_worker_logging, self._log_queue)
+
         self._pool = LaunchpadMultiProcessingPool(
             num_processes=concurrency,
-            initializer=self._initialize_worker_logging,
+            initializer=initializer_with_queue,
         )
         self.concurrency = concurrency
         self.max_pending_futures = max_pending_futures
         self.healthcheck_file = healthcheck_file
 
+    def _setup_queue_listener(self) -> QueueListener:
+        """Set up listener in main process to handle logs from workers."""
+        root_logger = logging.getLogger()
+        handlers = list(root_logger.handlers) if root_logger.handlers else []
+
+        return QueueListener(self._log_queue, *handlers, respect_handler_level=True)
+
+    @staticmethod
+    def _initialize_worker_logging(log_queue: multiprocessing.Queue[Any]) -> None:
+        """Initialize logging in worker process to send logs to queue.
+
+        With multiprocessing spawn context, subprocesses don't inherit
+        parent's stdout/stderr. We use a queue to send log records to
+        the main process which writes them to stdout for Docker/GCP.
+        """
+        root_logger = logging.getLogger()
+        root_logger.handlers.clear()
```

Comment:
Is this going to cause race conditions with one worker process wiping out the logger for another worker process? Not super clear to me why this has to be cleared.